patches/api/0323-Add-PiglinAngeredEvent.patch

```diff
@@ -0,0 +1,127 @@
++
++    private final LivingEntity target;
++    private final Reason reason;
++    private boolean cancel = false;
++
++    public PiglinAngeredEvent(@NotNull PiglinAbstract piglinAbstract, @NotNull LivingEntity target, @NotNull Reason reason) {
++        super(piglinAbstract);
++        this.target = target;
++        this.reason = reason;
++    }
++
++    /**
++     * Gets the target {@link LivingEntity} the {@link PiglinAbstract} is angered at.
++     * @return The target
++     */
++    @NotNull
++    public LivingEntity getTarget() {
++        return target;
++    }
++
++    /**
++     * Gets the reason the {@link PiglinAbstract} was angered.
++     * @return The reason
++     */
++    @NotNull
++    public Reason getReason() {
++        return reason;
++    }
++
++    @NotNull
++    @Override
++    public PiglinAbstract getEntity() {
++        return (PiglinAbstract) super.getEntity();
++    }
++
++    @Override
++    public boolean isCancelled() {
++        return cancel;
++    }
++
++    @Override
++    public void setCancelled(boolean cancel) {
++        this.cancel = cancel;
++    }
++
++    @NotNull
++    @Override
++    public HandlerList getHandlers() {
++        return handlers;
++    }
++
++    @NotNull
++    public static HandlerList getHandlerList() {
++        return handlers;
++    }
++
++    public enum Reason {
++        /**
++         * Another {@link PiglinAbstract} broadcasted anger towards the target
++         */
++        BROADCAST,
++        /**
++         * Another {@link PiglinAbstract} broadcasted universal anger
++         */
++        BROADCAST_UNIVERSAL,
++        /**
++         * A {@link org.bukkit.entity.Player} destroyed a block guarded by {@link PiglinAbstract}s
++         */
++        DESTROYED_GUARDED_BLOCK,
++        /**
++         * The {@link PiglinAbstract} is hunting a {@link org.bukkit.entity.Hoglin}
++         */
++        HUNT,
++        /**
++         * A {@link org.bukkit.entity.Player} opened a {@link org.bukkit.block.Chest} or destroys an {@link org.bukkit.entity.minecart.StorageMinecart} or {@link org.bukkit.entity.minecart.HopperMinecart}
++         */
++        OPENED_CONTAINER,
++        /**
++         * The {@link PiglinAbstract} is retaliating against an attacker
++         */
++        RETALIATION,
++        /**
++         * Unknown anger reason
++         */
++        UNKNOWN
++    }
++}
+diff --git a/src/main/java/org/bukkit/entity/PiglinAbstract.java b/src/main/java/org/bukkit/entity/PiglinAbstract.java
+index 87f4b7ad7c0a95a7123d142fa023c5da5c760341..a6c361da1dfa43aecf55ca90f097254817721eef 100644
+--- a/src/main/java/org/bukkit/entity/PiglinAbstract.java
++++ b/src/main/java/org/bukkit/entity/PiglinAbstract.java
+@@ -66,4 +66,6 @@ public interface PiglinAbstract extends Monster, Ageable {
+      */
+     @Deprecated
+     public void setBaby(boolean flag);
++
++    @org.jetbrains.annotations.Nullable LivingEntity getAngerTarget(); // Paper
```

Comment:
missing javadocs?

Reply:
don't forget to update the paper comments ;P

Reply:
🥴